internal/service/flexcluster/resource_test.go

```diff
@@ -53,25 +54,44 @@ func basicTestCase(t *testing.T) *resource.TestCase {
 
 func configBasic(projectID, clusterName string, terminationProtectionEnabled bool) string {
 	return fmt.Sprintf(`
-		resource "mongodbatlas_flex_cluster" "flex_cluster" {
+		resource "mongodbatlas_flex_cluster" "test" {
 			project_id = %[1]q
 			name       = %[2]q
 			provider_settings = {
 				backing_provider_name = "AWS"
 				region_name           = "US_EAST_1"
 			}
 			termination_protection_enabled = %[3]t
-		}`, projectID, clusterName, terminationProtectionEnabled)
+		}
+		data "mongodbatlas_flex_cluster" "test" {
+			project_id = mongodbatlas_flex_cluster.test.project_id
+			name       = mongodbatlas_flex_cluster.test.name
+		}
+		`, projectID, clusterName, terminationProtectionEnabled)
 }
 
 func checksFlexCluster(projectID, clusterName string, terminationProtectionEnabled bool) resource.TestCheckFunc {
 	checks := []resource.TestCheckFunc{checkExists()}
-	attributes := map[string]string{
+	attrMap := map[string]string{
 		"project_id":                     projectID,
 		"name":                           clusterName,
 		"termination_protection_enabled": fmt.Sprintf("%v", terminationProtectionEnabled),
```

Comment:
nit: not changed in this PR, this can also be written as: 
"termination_protection_enabled": strconv.FormatBool(terminationProtectionEnabled),